### PR TITLE
landing worker: send notification of failed transplants (bug 1649465)

### PR DESCRIPTION
**Pull request by zzzeid**
Add support for email notification when a landing job fails.

### CODE REVIEW

File: tests/test_landings.py [hunk 1]
```diff
@@ -185,3 +191,51 @@ def test_integrated_execute_job(
...
+    db,
+    s3,
+    mock_repo_config,
+    hg_server,
+    hg_clone,
+    treestatusdouble,
+    monkeypatch,
+    upload_patch,
+):
+    """Ensure that a failed landings triggers a user notification.
```
Comment (zzzeid) [inline tests/test_landings.py:207]:
```suggestion
    """Ensure that a failed landing triggers a user notification.
```

File: tests/test_landings.py [hunk 2]
```diff
@@ -10,6 +13,23 @@
 from landoapi.repos import Repo, SCM_LEVEL_3
 
 
+@pytest.fixture
+def upload_patch():
+    """A fixture that fake uploads a patch"""
```
Comment (zzzeid) [inline tests/test_landings.py:18]:
```suggestion
    """A fixture that fake uploads a patch."""
```

File: tests/test_notifications.py [hunk 3]
```diff
@@ -95,7 +95,13 @@ def test_notify_user_of_landing_failure(check_celery, app, celery_worker, smtp):
...
     # celery_worker fixture causes the test to cross threads.  We only ensure the
     # happy-path runs cleanly.
-    notify_user_of_landing_failure(Transplant(revision_order=["1"]))
+    transplant = Transplant(revision_order=["1"])
+    notify_user_of_landing_failure(
+        transplant.requester_email,
+        transplant.head_revision,
+        transplant.error,
+        transplant.request_id,
+    )
```
Comment (zalun) [inline tests/test_notifications.py:104]: Is creating the `Transplant` object is required?

Comment (zzzeid) [reply tests/test_notifications.py:104]: It is not required at all, indeed. I will update this in a follow up PR. This test needs to be rewritten though, as it's not really testing or doing much, as per the original comment above.

**Pull request closed without merging**